src/metabase/models/data_permissions.clj

```diff
@@ -16,6 +16,7 @@
 
 (t2/deftransforms :model/DataPermissions
   {:perm_type  mi/transform-keyword
+   :perm-type  mi/transform-keyword
```

Comment:
How does this work? Does it get translated to `perm_type` somewhere?

Reply:
It just defines the transformation to apply to the given column name. In this case, `perm_type` should be selected as a keyword.

But `:select [[:p.perm_type :perm-type]]` wasn't working unless I also add the transformation on `perm-type`, since I guess it applies it based on the aliased name in the result set. So I just added a transformation to both names. 🤷‍♂️ I don't know the exact implementation under the hood but this seems to work.